stan/math/opencl/matrix_cl.hpp

```diff
@@ -469,6 +469,13 @@ class matrix_cl<T, require_arithmetic_t<T>> : public matrix_cl_base {
             require_all_kernel_expressions_and_none_scalar_t<Expr>* = nullptr>
   matrix_cl<T>& operator=(const Expr& expression);
 
+  /**
+    Evaluates `this`. This is a no-op.
+    @return `*this`
+    */
+  const matrix_cl<T>& eval() const& { return *this; }
+  matrix_cl<T> eval() && { return std::move(*this); }
```

Comment:
Just to make sure I understand this syntax, this method is only used when the class is an rvalue hence &&?

Reply:
Yes.